cdap-docs/developers-manual/source/building-blocks/datasets/fileset.rst

```diff
@@ -130,3 +129,260 @@ See the Apache™ Twill®
+      .setExploreInputFormat("org.apache.hadoop.hive.ql.io.avro.AvroContainerInputFormat")
+      .setExploreOutputFormat("org.apache.hadoop.hive.ql.io.avro.AvroContainerOutputFormat")
+      .setTableProperty("avro.schema.literal", SCHEMA_STRING)
+
+You need to specify the SerDe, the input format, the output format, and any additional properties
+any of these may need as table properties. This is an experimental feature and only tested for
+Avro; see the :ref:`StreamConversion <examples-stream-conversion>` example for more details.
+
+======================
+TimePartitionedFileSet
+======================
+
+TimePartitionedFileSets are a special case (and in fact, a subclass) of PartitionedFileSets, where
+the partitioning is fixed to five integers representing the year, month, day of the month, hour of the day,
+and minute of a partition's time. For convenience, it offers methods to address the partitions by
+time instead of partition key or filter. The time is interpreted as milliseconds since the Epoch.
+
+The following convenience methods provide access to partitions by time instead of a partition key::
+
+  @Nullable
+  public TimePartition getPartitionByTime(long time);
+
+  public Set<TimePartition> getPartitionsByTime(long startTime, long endTime);
+
+  @Nullable
+  public TimePartitionOutput getPartitionOutput(long time);
+
+Essentially, these methods behave the same as if you convert the time arguments into partition keys and
+call the corresponding methods of ``PartitionedFileSet`` with the resulting partition keys. In addition:
+
+- The returned partitions have an extra method to retrieve the partition time as long.
+- The start and end time of ``getPartitionsByTime()`` do not correspond directly to a single partition filter,
+  but to a series of partition filters. For example, to retrieve the partitions between November 2014 and
+  March 2015, you need two partition filters: One for the months of November to December of 2014, and one
+  for January through March of 2015. This method converts the given time range into the corresponding set
+  of partition filters, retrieves the partitions for each filter, and returns the superset of all these
+  partitions.
+
+Using TimePartitionedFileSets in MapReduce
+==========================================
+
+Using time partitioned file sets in MapReduce is similar as for partitioned file sets; however, instead of
+setting an input partition filter and an output partition key, you configure an input time range and an
+output partition time in the ``beforeSubmit()`` of the MapReduce::
+
+    TimePartitionedFileSetArguments.setInputStartTime(inputArgs, startTime);
+    TimePartitionedFileSetArguments.setInputEndTime(inputArgs, endTime);
+
+and::
+
+    TimePartitionedFileSetArguments.setOutputPartitionTime(outputArgs, partitionTime);
+
+Or, you could achieve the same result by specifying the input time range and the output partition time
```

Comment:
Putting this in parentheses is a little weird, because the next paragraph including the example belong to it. Not sure how to put a quoted example in parentheses... I am adopting your text, but keeping it a single paragraph. Does that work?